### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by ZuGao**
## Summary

Added documentation:
* Installation guide;
* How to create a release;
* Examples.

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog

### CODE REVIEW

File: docs/antora-build.mk [hunk 1]
```diff
@@ -1,4 +1,5 @@
 docs_out_dir := ./.public
+docs_moduleroot_dir := docs/modules/ROOT
```
Comment (ccremer) [inline docs/antora-build.mk:2]: Why this change? doesn't seem to be used.

**Comment by Kidswiss** [comment]
LGTM

But shouldn't this PR also have the label `documentation`?

**Review by Kidswiss — APPROVED** [review]

**Comment by ZuGao** [comment]
> LGTM
> 
> But shouldn't this PR also have the label `documentation`?

yes you are right, updated the label.

File: docs/antora-build.mk (hunk 1 above)
Comment (ZuGao) [reply docs/antora-build.mk:2]: It is used in [Makefile](https://github.com/vshn/provider-exoscale/blob/master/Makefile#L70). Unfortunately rewriting the commit history is not always perfect.

**Review by ccremer — APPROVED** [review]
